README.md

```diff
@@ -1,17 +1,119 @@
 
-This template repository defines the typical structure for a Griptape tool and is meant to be used as a starting point when developing new tools. To create a tool using this template, click the **"Use this template"** button at the top of the repository.
+Via github web page:
 
-### Steps for creating a new Griptape tool:
+Click on `Use this template`
+
+![](https://docs.github.com/assets/cb-36544/images/help/repository/use-this-template-button.png)
+
+
+Via `gh`:
+
+```
+$ gh repo create griptape-extension-name -p griptape/griptape-extension-template
+```
+
+## What is a Griptape Extension?
+
+Griptape Extensions can add new functionality to the [griptape framework](https://github.com/griptape-ai/griptape), such as new Tools, Drivers, Tasks, or Structures.
+
+This repository provides a recommended structure for organizing your extension code, as well as helpful tools for testing and development.
+
+## Extension Structure
+
+The template repository is structured as follows:
+
+```bash
+tree -I __init__.py -I __pycache__
+
+├── griptape
+│   └── extension_name # Name whatever you want
+│       └── tools
+│           └── reverse_string
+│               └── tool.py
+            ...more directories for other interfaces (drivers, tasks, structures, etc)...
+└── tests
+    └── unit
+        └── tools
+            └── test_reverse_string_tool.py
+├── examples
+│   └── example.py # Example usage of the extension 
+├── LICENSE # Choose the appropriate license
+├── Makefile # Contains useful commands for development
+├── pyproject.toml # Contains the project's metadata
+├── README.md #  Describes the extension and how to use it
+```
+
+## Development
+
+### Poetry
+
+This project uses [Poetry](https://python-poetry.org/) for dependency management.
+It is recommended to configure Poetry to use [in-project](https://python-poetry.org/docs/configuration/#virtualenvsin-project) virtual environments:
+
+```bash
+poetry config virtualenvs.in-project true
+```
+
+This will create a `.venv` directory in the project root, where the virtual environment will be stored.
+This ensures that the virtual environment is always in the same location, regardless of where the project is cloned.
+
+### Useful Commands
+
+#### Installing Dependencies
+
+```bash
+make install
+```
+
+#### Running Tests
+
+```bash
+make test
+```
+
+#### Running Checks (linting, formatting, etc)
+
+```bash
+make check
+```
+
+#### Running Formatter
+
+```bash
+make format
+```
+
+#### Running Example
+
+1. Set appropriate environment variables. The example requires an `OPENAI_API_KEY` environment variable to be set.
+2. Run the example:
+
+```bash
+poetry run python examples/example.py
+```
+
+### Installing in Other Projects
+
+Extensions are designed to be shared and installed in projects that can benefit from them.
+
+Although publishing to [PyPI](https://pypi.org/) is an option, it is beyond the scope of this template. Instead, extensions can be installed directly from the repository.
```

Comment:
I'd remove the PyPi comment, because it gives the reader a sense of dread (n.b.: I am always sensing dread). Be optimistic: `The easiest way to include your extension into an existing project is to install directly from the repository, like so:`

...

later:
`Advanced customers may seek to publish their extensions to PyPi. Those instructions are beyond the scope of this README.`